app/components/portfolios/details_component.rb

```diff
@@ -54,10 +54,39 @@ def all_subprojects
       all_descendants.project
     end
 
+    def render_sub_status_bar?
+      # When none of the descendants have a status set, there's nothing to show and we
+      # will return false.
+      sub_statuses.keys.any?(&:present?)
+    end
+
+    def sub_statuses_with_percentages
+      return @status_percentage if @status_percentage
+
+      statuses = sub_statuses
+      total = statuses.values.sum
+
+      @status_percentage = statuses.map do |code, count|
+        percentage = ((count.to_f / total) * 100).round
```

Comment:
I think using either `fdiv` or `100.0` is cleaner.
Also wanted to suggest rounding to 1-2 decimal digits, for example for having only 3 projects with different statuses leading to 3 times 33%, but I don't see a visible artifact in the progress bar

```suggestion
        percentage = (count.fdiv(total) * 100).round
```

```suggestion
        percentage = (count * 100.0 / total).round
```